crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -500,8 +500,10 @@ fn derive_label(input: DeriveInput, label_type: Ident) -> TokenStream2 {
     };
     let crate_path: Path = syn::parse(path_str.parse::<TokenStream>().unwrap()).unwrap();
 
+    let (impl_generics, ty_generics, where_clause) = input.generics.split_for_impl();
+
     quote! {
-        impl #crate_path::#label_type for #ident {
+        impl #impl_generics #crate_path::#label_type #ty_generics for #ident #where_clause {
```

Comment:
```suggestion
        impl #impl_generics #crate_path::#label_type for #ident #ty_generics #where_clause {
```
See [`syn` docs](https://docs.rs/syn/1.0.60/syn/struct.Generics.html#method.split_for_impl). Maybe there should be a test for this.